webapp/javascript/components/FlameGraph/FlameGraphComponent/DiffLegendPaletteDropdown.tsx

```diff
@@ -3,45 +3,87 @@ import Dropdown, { MenuItem, MenuButton } from '@ui/Dropdown';
 import Icon from '@ui/Icon';
 import dropdownStyles from '@ui/Dropdown.module.scss';
 import cx from 'classnames';
+import useResizeObserver from '@react-hook/resize-observer';
 import { ColorBlindPalette, DefaultPalette } from './colorPalette';
 import DiffLegend from './DiffLegend';
 import CheckIcon from '../../CheckIcon';
+
 import styles from './DiffLegendPaletteDropdown.module.css';
 
 const paletteList = [DefaultPalette, ColorBlindPalette];
 
 export const DiffLegendPaletteDropdown = (props) => {
   const { palette = DefaultPalette, onChange } = props;
+  const legendRef = React.useRef();
+  const showMode = useSizeMode(legendRef);
+
   return (
-    <Dropdown
-      label="Select a palette"
-      menuButton={
-        <MenuButton
-          className={cx(
-            // eslint-disable-next-line
-            dropdownStyles.dropdownMenuButton,
-            styles.diffPaletteDropdown
-          )}
-        >
-          <DiffLegend palette={palette} />
-        </MenuButton>
-      }
-      onItemClick={(e) => onChange(e.value)}
-    >
-      {paletteList.map((p) => (
-        <MenuItem key={p.name} value={p}>
-          <div>
-            <label>{p.name}</label>
-            <div className={styles.dropdownItem}>
-              <DiffLegend palette={p} />
-
-              {p === palette ? <CheckIcon /> : null}
+    <div ref={legendRef} className={styles.dropdownWrapper}>
+      <Dropdown
+        label="Select a palette"
+        menuButton={
+          <MenuButton
+            className={cx(
+              // eslint-disable-next-line
+              dropdownStyles.dropdownMenuButton,
+              styles.diffPaletteDropdown
+            )}
+          >
+            <DiffLegend palette={palette} showMode={showMode} />
+          </MenuButton>
+        }
+        onItemClick={(e) => onChange(e.value)}
+      >
+        {paletteList.map((p) => (
+          <MenuItem key={p.name} value={p}>
+            <div>
+              <label>{p.name}</label>
+              <div className={styles.dropdownItem}>
+                <DiffLegend palette={p} showMode={showMode} />
+
+                {p === palette ? <CheckIcon /> : null}
+              </div>
             </div>
-          </div>
-        </MenuItem>
-      ))}
-    </Dropdown>
+          </MenuItem>
+        ))}
+      </Dropdown>
+    </div>
   );
 };
 
+/**
+ * TODO: unify this and toolbar's
```

Comment:
I'm fine leaving this as a todo, but as a follow up after we merge we should create an issue so we don't forget about this